lib/rspec/core/example.rb

```diff
@@ -92,15 +92,32 @@ def inspect_output
         inspect_output
       end
 
-      # Returns the argument that can be passed to the `rspec` command to rerun this example.
-      def rerun_argument
-        loaded_spec_files = RSpec.configuration.loaded_spec_files
+      # Returns the location-based argument that can be passed to the `rspec` command to rerun this example.
+      def location_rerun_argument
+        @location_rerun_argument ||= begin
+          loaded_spec_files = RSpec.configuration.loaded_spec_files
```

Comment:
I'm apprehensive, too, and would love to find a way to do away with the global state, but it's definitely not an easy change.  Many, many call sites would have to change to accept `RSpec.configuration` and/or `RSpec.world` in order to pass them on.  I looked into a bit at one point and concluded it wasn't worth the ROI, but you can certainly take a stab at it :).